test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -622,24 +622,25 @@ private static String groupName(ThreadGroup threadGroup) {
      * recursive shuffling behavior can be made by passing in the names of fields which
      * internally should stay untouched.
      */
-    public static XContentBuilder shuffleXContent(XContentBuilder builder, Set<String> exceptFieldNames) throws IOException {
+    public static XContentBuilder shuffleXContent(XContentBuilder builder, String... exceptFieldNames) throws IOException {
         BytesReference bytes = builder.bytes();
         XContentParser parser = XContentFactory.xContent(bytes).createParser(bytes);
         // use ordered maps for reproducibility
         Map<String, Object> shuffledMap = shuffleMap(parser.mapOrdered(), exceptFieldNames);
-        XContentBuilder jsonBuilder = XContentFactory.contentBuilder(builder.contentType());
+        XContentBuilder jsonBuilder = XContentFactory.contentBuilder(builder.contentType()).prettyPrint();
```

Comment:
> maybe only set prettyPrint if the original had it set?
I was trying to do that, but only possible when adding getters for the pretty flag to XContentBuilder and XContentGenerator. I didn't want to do that originally, but now I think I'll just add them, doesn't hurt to have those.